### PR TITLE
Add AddSequenceDiagram

### LINKED ISSUE #86: Add AddSequenceDiagram
Add AddSequenceDiagram

### PR DESCRIPTION
**Pull request by SSirMentos**
Add AddSequenceDiagram
close #86

### CODE REVIEW

File: docs/diagrams/AddSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+activate AddressBookParser
+
+create AddCommandParser
+AddressBookParser -> AddCommandParser
+activate AddCommandParser
+
+AddCommandParser --> AddressBookParser
+deactivate AddCommandParser
+
+AddressBookParser -> AddCommandParser : parse(John)
```
Comment (M-kew) [inline docs/diagrams/AddSequenceDiagram.puml:30]: If 'John' is supposed to represent the command, then it is better to write down ("add n/ John Doe p/...) instead? What do you think.

File: docs/diagrams/AddSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+[-> LogicManager : execute("add n/John Doe p/...")
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand("add n/John Doe p/...")
+activate AddressBookParser
+
+create AddCommandParser
+AddressBookParser -> AddCommandParser
+activate AddCommandParser
+
```
Comment (M-kew) [inline docs/diagrams/AddSequenceDiagram.puml:26]: Assuming 'John' is an object, then I think you might need to add a line 

create Person
AddCommandParser -> Person : new Person("John Doe")
activate Person

File: docs/diagrams/AddSequenceDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+create AddCommand
+AddCommandParser -> AddCommand
+activate AddCommand
+
+AddCommand --> AddCommandParser :
+deactivate AddCommand
+
+AddCommandParser --> AddressBookParser : a
```
Comment (M-kew) [inline docs/diagrams/AddSequenceDiagram.puml:40]: if 'a' is the return value, you need to return the CommandResult if I'm not wrong. What do you think?

File: docs/diagrams/AddSequenceDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+activate Model
+
+Model --> AddCommand
+deactivate Model
+
+create CommandResult
+AddCommand -> CommandResult
+activate CommandResult
+
+CommandResult --> AddCommand
```
Comment (M-kew) [inline docs/diagrams/AddSequenceDiagram.puml:64]: Might need to add generaterResult() for the creation of CommandResult(). By explicitly showing the generateResult() method, you clarify that AddCommand is not just passing an existing result but is actively generating the CommandResult. This highlights that there is a process involved in creating the result after executing the addPerson() operation.

**Comment by M-kew** [comment]
I have given some comments for your UML diagram. Please take a look and let me know what you think. Need to resolve before pushing.

File: docs/diagrams/AddSequenceDiagram.puml (hunk 1 above)
Comment (SSirMentos) [reply docs/diagrams/AddSequenceDiagram.puml:30]: no John is the Person Object

File: docs/diagrams/AddSequenceDiagram.puml (hunk 2 above)
Comment (SSirMentos) [reply docs/diagrams/AddSequenceDiagram.puml:26]: that make sense!

**Pull request merged**